### PR TITLE
fix git annotations bug

### PR DESCRIPTION
**Pull request by shuyaqian**
Signed-off-by: shuyaqian 717749594@qq.com

#### What it does
fix [#9998](https://github.com/eclipse-theia/theia/issues/9998)

#### How to test
1. 'alt + b' to execute git.editor.toggle.annotations command
2. when characters other than English exist，width calculation is incorrect

#### Review checklist

- [x] As an author, I have thoroughly tested my changes and carefully followed [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#requesting-a-review)

#### Reminder for reviewers

- As a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)

### CODE REVIEW

File: packages/git/src/browser/blame/blame-decorator.ts [hunk 1]
```diff
@@ -200,23 +186,41 @@ export class BlameDecorator implements monaco.languages.HoverProvider {
...
+        backgroundColor: 'var(--theia-gitlens-gutterBackgroundColor)',
+        height: '100%',
+        display: 'inline-block',
+        overflow: 'hidden',
+        textOverflow: 'ellipsis'
+    };
 
-    export const defaultGutterStyles = <CSSStyleDeclaration>{
-        width: `${maxWidth}ch`,
+    export const defaultGutterAfterStyles = <CSSStyleDeclaration>{
```
Comment (colin-grant-work) [inline packages/git/src/browser/blame/blame-decorator.ts:200]: Removing exports from this namespace (defaultGutterStyles, maxWidth, continuationStyle, highlightStyle) is a breaking change. Please add an entry to the changelog reflecting the changes.

**Review by colin-grant-work — COMMENTED** [review]
This is an improvement over the old code, but I was still able to get a bad result when the message overflowed:

image

Since not all reviewers may be comfortable with non-Latin characters, you may want to include a test commit as part of this PR with a message that includes non-Latin characters, and add instructions in your PR description for where to find those messages.

**Comment by shuyaqian** [comment]
> This is an improvement over the old code, but I was still able to get a bad result when the message overflowed:
> 
> image
> 
> Since not all reviewers may be comfortable with non-Latin characters, you may want to include a test commit as part of this PR with a message that includes non-Latin characters, and add instructions in your PR description for where to find those messages.

done.

**Comment by colin-grant-work** [comment]
With the recent changes, this is looking good to me:
image

Please squash the commits, and I'll approve.

**Comment by msujew** [comment]
@colin-grant-work Even with the recent changes I'm still experiencing the issue you mentioned previously:

image

**Comment by colin-grant-work** [comment]
> @colin-grant-work Even with the recent changes I'm still experiencing the issue you mentioned previously:

Hm... thanks for checking. Gotta try more scripts... :-)

**Comment by colin-grant-work** [comment]
It looks like Japanese, Korean, and Arabic are all causing problems:

image

I used these messages: 
Courtesy of @msujew 
> わたしにほんごがすこししかはなせません
> 한국어에서 Git 주석이 어떻게 보이는지 알고 싶습니다.
> أريد أن أعرف كيف تبدو التعليقات التوضيحية باللغة الكورية.

The Arabic message is also displaying the end rather than the beginning - but handling RTL text correctly may be beyond the scope of this PR.

**Comment by shuyaqian** [comment]
> It looks like Japanese, Korean, and Arabic are all causing problems:
> 
> image
> 
> I used these messages:
> 
> > わたしにほんごがすこししかはなせません
> > Courtesy of @msujew
> 
> > 한국어에서 Git 주석이 어떻게 보이는지 알고 싶습니다.
> > أريد أن أعرف كيف تبدو التعليقات التوضيحية باللغة الكورية.
> 
> The Arabic message is also displaying the end rather than the beginning - but handling RTL text correctly may be beyond the scope of this PR.

I have fixed it.

**Comment by shuyaqian** [comment]
image

**Comment by colin-grant-work** [comment]
@shuyaqian, it looks like there's a problem with a reference to a non-existent namespace field.

**Comment by shuyaqian** [comment]
> @shuyaqian, it looks like a problem with a reference to a non-existent namespace field is crashing the build.

That's ok now.

**Comment by shuyaqian** [comment]
> The style specifications now look robust, and I can't identify any cases in which the time message overflows.

You can test it like this
img

**Review by msujew — APPROVED** [review]
Looks good to me as well. Using different scripts/languages I could no longer reproduce the issue 👍

**Pull request merged**